### DISCUSSION TITLE
Create a discord server (Edit: invite link available)

### DISCUSSION THREAD

**Discussion by iAndyHD3** [axmolengine/axmol#1181] [General]
Edit: [invite link](https://discord.gg/dJR7SPNRzm)
--

It would be great if axmol had a discord server for general discussions and small questions that don't necessarily need creating a specific issue. discord would be incredibly helpful in building the community that was lost in Cocos2dx, potentially leading to an increase in the size of the axmol community. In fact, some users have already been discussing axmol in the official Cocos2d Discord server (as seen in the attached image).

![image](https://user-images.githubusercontent.com/54410739/233805475-83825684-371c-4089-8261-5c2c13670d90.png)

**Comment by rh101** [comment]
It may seem like a good idea to have a presence on services like Discord, but in reality, it really isn't.  

If a question needs to be asked, then the most appropriate place is a forum, such as this "Discussions" section of the github project.  

With Discord, there are a lot of issues, some which may not be immediately obvious. For instance:

- The live nature of it would make moderation much harder.  Bad content will reflect badly on the project, and is often the cause of conflict between users.
- The previous point would be solved by moderation, but that in itself is a problem, because moderation requires users to be active, which means devoting time and resources to that role.  This is an open source project, so it is unreasonable to expect people to place themselves in this role.
- Even if people were to put their hands up to be moderators, then the moderators themselves would require some kind of oversight, so someone always has to be active to ensure that the moderators themselves are not the cause of issues.  Moderators are just people, and people make mistakes; that's just the reality of it. 

Now let's get to the users and the content.

- On Discord, there would be a lot of "noise" in between useful pieces of information, and the ratio is significant. So, it would be tough to find the information you need, or to even follow it.  Now, with a forum, the questions and answers are contained together in a thread, even if there were unrelated answers in there, the amount of "noise" is almost insignificant.
- Users on Discord are just people, and some people tend to feel like their time is worth more than other people's. This results in problems, such as users constantly re-posting the same question, or insisting that their question be answered in the time frame they want, again forgetting that there are no support people being paid to help them.  Also, there is a bigger issue of unsolicited direct messages to other users.  This is a pain point, and would end with users becoming annoyed, and blocking users, or simply ignoring or leaving the service altogether, and defeats the purpose of having it there in the first place.

Where Discord is great is as a communication tool for transient information, something that doesn't need to last long, which is why it's great for being used for random posts or as a voice chat tool for past-times like gaming.  

For information that should be persisted, and easily accessible, a place like this forum would be the best place for such things.

**Reply by DelinWorks** [reply]
While I do mostly agree with what you've said. except for the users getting frustrated part because that rarely happens and that can be in any community and it includes communications through GitHub too, simply because they didn't find their answer listed and it makes them frustrated.. should they really be using a C++ engine in the beginning? it doesn't hurt to try! we're a small community and the only way to build this community is by having other communication methods. This community growth would impact the engine positively with a lot of contributors and new comers! I've been with discord servers relating to game engines and the moderation there isn't really needed, people just ask questions and get answers and discuss. that's it! If discord communication really needs moderation then the same could be said for GitHub.

**Reply by rh101** [reply]
> While I do mostly agree with what you've said. except for the users getting frustrated part because that rarely happens 

You may not have experienced being the target of users with ill intent, but I can assure you that it actually happens often, and given that there is direct messaging support on Discord, the spam is much worse.  People with lots of time on their hands can get nasty.

> and that can be in any community and it includes communications through GitHub too, simply because they didn't find their answer listed and it makes them frustrated.. should they really be using a C++ engine in the beginning?

Github is essentially the moderator, and there are measures in place to both limit and address abuse, so it doesn't fall on the project maintainers to handle this.  Again, that is not the case with Discord, as the moderation falls on the server maintainer and any moderators assigned to it.  The two services are not directly comparable like this.

> it doesn't hurt to try! we're a small community and the only way to build this community is by having other communication methods. This community growth would impact the engine positively with a lot of contributors and new comers! I've been with discord servers relating to game engines and the moderation there isn't really needed, people just ask questions and get answers and discuss. that's it! 

Again, I strongly recommend against it.  The experience using one server does not mean it will translate to another, plus things may be going on behind the scenes that you may not be aware of (such as moderators taking care of issues before they're even noticed). 

Questions and answers should be in a place where they are persistent and easily accessible in order to help future users find that information. A chat service like Discord was never intended for such things.

Additionally, Discord is not a marketing tool, so it doesn't make sense to assume it will give this project more exposure.  The people that join a Discord server would already be part of whatever community it's associated with.

For me personally, I definitely would not want to be a part of such a thing, and while I do use Discord, it is for unrelated things (nothing to do with coding, work, or anything of importance).  I did join the Cocos2d-x Discord at one point, and even answers questions a few times, but I went through the problems listed above in my initial post, such as receiving constant unsolicited direct messages from users.  In the end I left their server, and don't particularly wish to deal with that again.

> If discord communication really needs moderation then the same could be said for GitHub.

As I mentioned earlier, it's not directly comparable, since Github is a commercial company that has policies and resources in place to deal with moderation, and is not the case with a Discord server, which is community based.

Projects such as this come from necessity.  If someone wants to make a game, or learn how to make one, then the way they probably start is by a web search.  The more places this engine is referenced, the more likely people will find their way here.  That also doesn't mean spamming other sites with links to Axmol.  As an example, the most natural way is if the engine is mentioned in places for a specific reason, such as a blog article related to Axmol usage, or linking Axmol on other general game development sites when users ask for game engine recommendations.

**Reply by DelinWorks** [reply]
Thanks for clarifying! I get what you mean now.

**Reply by iAndyHD3** [reply]
Thanks for providing a good answer explaining everything in detail. One last question, how would you feel about an unofficial discord managed by other people (e.g my team) and having the link posted here? The server would be **100% community based and I don't expect any developer to join**. We understood all of your concerns about discord but understand that me and my team would still like to give it a try (simply see the amount of upvotes).

I insist, for random discussions or small issues or literally any topic a discord can be pretty useful.

**Reply by rh101** [reply]
I don't really have an opinion on that, as there is nothing stopping anyone from creating a Discord server for any reason, so it's entirely your choice.  The only thing to be aware of is making it clear to users that it is a community-run server, and not officially tied to this project.  

@halx99 is the one who (thankfully) created this fork and maintains this project, so it is his opinion that matters regarding this.

**Reply by iAndyHD3** [reply]
You are right that nobody stops me, but I want to ask first because I obviously don't want to do something unofficial (that affects the project and) that maintainers don't want at all
We would make it clear that we are community based.

**Reply by halx99** [reply]
I think you can try.

**Comment by iAndyHD3** [comment]
we finished setting up one, [invite link](https://discord.gg/dJR7SPNRzm)

**Reply by DelinWorks** [reply]
Nice!
